File: solution files/Javascript/part-01/task-01-blueprint-decoder/blueprint_decoder.js
const fs = require("fs");

/**
 * Reads a blueprint file and extracts all secrets marked between {* and *}
 * Example: League transmission contains {* EMERGENCY_PROTOCOL: NIGHTFALL_SEQUENCE_ACTIVE *}
 * Extracts: "EMERGENCY_PROTOCOL: NIGHTFALL_SEQUENCE_ACTIVE" (without markers)
 * Uses regex pattern with match() or matchAll()
 * @param {string} filename - Path to the blueprint file
 * @returns {Array<string>} Array of extracted secret strings
 */
function decodeBlueprint(filename) {
    const content = fs.readFileSync(filename, "utf8");

    // Use regex to find all secrets between {* and *}
    const pattern = /\{\* (.*?) \*\}/g;
    const matches = [...content.matchAll(pattern)];
    const secrets = matches.map((match) => match[1]);

    return secrets;
}

// Test the decoder
const secrets = decodeBlueprint("blueprint-data.txt");
console.log(`Found ${secrets.length} secret(s):`);
secrets.forEach((secret, index) => {
    console.log(`${index + 1}. ${secret}`);
});

module.exports = { decodeBlueprint };

File: solution files/Python/part-01/task-01-blueprint-decoder/blueprint_decoder.py
import re

# Function to read a blueprint file and extract all secrets marked between {* and *}
# Example: League Blueprint contains {* AGENT_CODENAME: SHADOWMIND *}
# Should extract: "AGENT_CODENAME: SHADOWMIND" (without the markers)
# Uses regex pattern to find all occurrences
# Returns a list of extracted secrets
def decode_blueprint(filename):
    with open(filename, "r") as file:
        content = file.read()

    # Use regex to find all secrets between {* and *}
    pattern = r"\{\* (.*?) \*\}"
    secrets = re.findall(pattern, content)
    return secrets

# Test the decoder
if __name__ == "__main__":
    secrets = decode_blueprint("blueprint-data.txt")
    print(f"Found {len(secrets)} secret(s):")
    for i, secret in enumerate(secrets, 1):
        print(f"{i}. {secret}")
